### PR TITLE
Student Application Dialog

### LINKED ISSUE #48: Student Application Dialog
Create a dialog to view the details of the application of the student before approving or rejecting.

### PR DESCRIPTION
**Pull request by 29-29**
Created a dialog to review the details of the application of a student before approving or rejecting.

### CODE REVIEW

File: src/components/admin/DataTable.vue [hunk 1]
```diff
@@ -1,60 +1,49 @@
 <template>
   <q-table
+    hide-bottom
```
Comment (fencim) [inline src/components/admin/DataTable.vue]: adding this removes the pagination preventing from viewing the content on the next page

File: src/components/admin/ActionButtons.vue [hunk 2]
```diff
@@ -1,19 +1,79 @@
...
+              payload: { ...props.props, data: { ...props.props.data } },
+            },
+          });
+        },
+      },
+      {
+        label: 'No',
+      },
+    ],
+  });
```
Comment (fencim) [inline src/components/admin/ActionButtons.vue]: You may remove this, as this is temporary implementation. View Application is sufficient.

File: src/components/admin/ActionButtons.vue [hunk 3]
```diff
@@ -1,19 +1,79 @@
...
-  <q-td :props="props" class="text-right">
-    <q-btn color="green" class="q-mx-sm">Approve</q-btn>
+  <div class="flex justify-center">
     <q-btn
+      round
+      flat
+      color="green"
+      class="q-mx-sm"
+      icon="check"
+      @click="approveApplication"
```
Comment (fencim) [inline src/components/admin/ActionButtons.vue]: You may want to use this as to `viewApplication` which leads to `approveApplication`

Comment (fencim) [inline src/components/admin/ActionButtons.vue]:
```suggestion
      @click="viewApplication"
```

File: src/components/admin/ActionButtons.vue [hunk 4]
```diff
@@ -1,19 +1,79 @@
...
-      @click="TheDialogs.emit({ type: 'applicationRejectDialog', arg: {} })"
-      >Reject</q-btn
+      @click="rejectApplication"
     >
-  </q-td>
+      <q-tooltip>Reject</q-tooltip>
+    </q-btn>
+    <q-btn round flat icon="more_horiz" @click="viewApplication">
+      <q-tooltip>View Application</q-tooltip>
+    </q-btn>
```
Comment (fencim) [inline src/components/admin/ActionButtons.vue]: You can remove this button by keeping the approve button to view details

File: src/components/admin/DataTable.vue [hunk 5]
```diff
@@ -63,32 +52,37 @@
...
+//   },
+//   {
+//     label: 'Approved',
+//     value: 'approved',
+//   },
+//   {
+//     label: 'Rejected',
+//     value: 'rejected',
+//   },
+// ] as { label: string; value: IRequest['status'] }[];
```
Comment (fencim) [inline src/components/admin/DataTable.vue]: remove commented codes

**Review by fencim — DISMISSED** [review]

**Review by eyesoreee — APPROVED** [review]

**Review by fencim — APPROVED** [review]
Good job

**Pull request merged**